imagegw/shifter_imagegw/api.py

```diff
@@ -211,6 +216,79 @@ def pull(system, imgtype, tag):
     return jsonify(create_response(rec))
 
 
+# Import image
+# This will import the requested image from a file path on the system.
+@app.route('/api/doimport/<system>/<imgtype>/<path:tag>/', methods=["POST"])
+def doimport(system, imgtype, tag):
+    """ 
+    Pull a specific image and tag for a systems. 
+    """
+    if imgtype == "docker" and tag.find(':') == -1:
+        tag = '%s:latest' % (tag)
+
+    auth = request.headers.get(AUTH_HEADER)
+    app.logger.debug("inside doimport")
+    data = {}
+    try:
+        rqd = request.get_data()
+        app.logger.debug("rqd '%s'" % (rqd))
+        if rqd is not "" and rqd is not None:
+            data = json.loads(rqd)
+            app.logger.debug("data '%s'" % (data))
+    except:
+        app.logger.warn("Unable to parse doimport data '%s'" %
+                        (request.get_data()))
+        pass
+
+    memo = "import system=%s imgtype=%s tag=%s" % (system, imgtype, tag)
+    app.logger.debug(memo)
+    i = {'system': system, 'itype': imgtype, 'tag': tag}
+    app.logger.debug("print data '%s'" % (data))
+    #Check for path to import file
+    if 'filepath' in data:
+        i['filepath'] = data['filepath']
+    else:
+        raise OSError("filepath required for direct image import")
+    if 'format' in data:
+        i['format'] = data['format']
+    else:
+        raise OSError("file type (e.g. squashfs) required for direct image import")        
+    #Check for list of allowed users or groups
+    if 'allowed_uids' in data:
+        # Convert to integers
+        i['userACL'] = map(lambda x: int(x),
+                           data['allowed_uids'].split(','))
+    if 'allowed_gids' in data:
+        # Convert to integers
+        i['groupACL'] = map(lambda x: int(x),
+                            data['allowed_gids'].split(','))
+    try:
+        app.logger.debug(i)
+        session = mgr.new_session(auth, system)
+        app.logger.debug(session)
+        #only allowed users can import images
+        user = session['user']
+        if 'ImportUsers' not in config.keys():
+            raise OSError("User image import from file disabled.")
+        iusers = config['ImportUsers']
+        #If ImportUsers is None, no one can do this
+        if iusers == "None" or iusers == "none":
```

Comment:
iusers.lower() == "none".  May also want to check for iusers is None.

Reply:
It's already checking for None or are you talking about another variable?